### PR TITLE
Allow register/degregister istioctl commands to work with .pem

### PR DESCRIPTION
**Pull request by esnible**
This PR adds the `istioctl kube-inject` fix to `istioctl register` and `deregister`.
See PR https://github.com/istio/istio/pull/5300

See Issue https://github.com/istio/istio/issues/5255

### CODE REVIEW

File: istioctl/cmd/istioctl/register.go [hunk 1]
```diff
@@ -40,14 +40,14 @@ var (
 				}
 				portsList[i] = p
 			}
-			log.Infof("Registering for service '%s' ip '%s', ports list %v",
+			log.Debugf("Registering for service '%s' ip '%s', ports list %v",
```
Comment (ayj) [inline istioctl/cmd/istioctl/register.go]: cc @ldemailly

**Comment by linsun** [comment]
@esnible note: we want this change in 0.8 first, think you can update to merge to 0.8 easily.  You may also need to merge https://github.com/istio/istio/pull/5300 to 0.8 first too.  cc @ayj

**Comment by linsun** [comment]
+1 to what @rshriram said but could be a separate PR.   @esnible let me know if you will pull in Jason's change so we can get the kube-inject cmd fixed in 0.8 with ibm cluoud.

File: istioctl/cmd/istioctl/register.go (hunk 1 above)
Comment (ldemailly) [reply istioctl/cmd/istioctl/register.go]: Leave as it was / why the change ?

File: istioctl/cmd/istioctl/register.go [hunk 2]
```diff
@@ -40,14 +40,14 @@ var (
...
 				svcName, ip, portsList)
 			if svcAcctAnn != "" {
 				annotations = append(annotations, fmt.Sprintf("%s=%s", kube.KubeServiceAccountsOnVMAnnotation, svcAcctAnn))
 			}
-			log.Infof("%d labels (%v) and %d annotations (%v)",
+			log.Debugf("%d labels (%v) and %d annotations (%v)",
```
Comment (ldemailly) [inline istioctl/cmd/istioctl/register.go]: Ditto

**Comment by esnible** [comment]
If register/deregister are no longer used for VMs we should get rid of those commands, not fix them.  I don't have a use case.  I was fixing these commands while researching a fix I discovered these commands had the same flaw as `kube-inject`.

The PR https://github.com/istio/istio/pull/5300 was merged yesterday.  I am no longer the On Call and have no power to merge it into 0.8.

**Comment by ldemailly** [comment]
0.8 is where the normal process still takes place and PRs should go.

Mesh expension should still work

**Comment by esnible** [comment]
Closing because there is no use case for these commands and no urgent need to remove them.

**Comment by ldemailly** [comment]
Mesh expension is the use case, if the commands got broken they need a fix

**Pull request closed without merging**